docs/azure/role-assignments.md

```diff
@@ -0,0 +1,261 @@
+
+## Default built-in role assignments
+
+When you add an Azure resource to the [app model](xref:dotnet/aspire/app-host#terminology), it's assigned default roles. If a resource depends on another resource, it inherits the same role assignments as the referenced resource unless explicitly overridden.
+
+Consider a scenario where an API project resource references an [Azure Search](../azureai/azureai-search-document-integration.md) resource. The API project is given the default role assignments, as shown in the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.AddAzureSearch("search");
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithReference(search);
+```
+
+In the example code, the `api` project resource depends on the Azure `search` resource, meaning it references the `search` resource. By default, the `search` resource is assigned the following built-in roles:
+
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataContributor?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchServiceContributor?displayProperty=nameWithType>
+
+These role assignments allow the API project to read and write data to the Azure Search resource, and manage it. However, this behavior might not always be desirable. For instance, you might want to restrict the API project to only read data from the Azure Search resource.
+
+## Override default role assignments
+
+<!-- TODO: Add xref links when available for the WithRoleAssignments API. -->
+
+To override the default role assignment, use the `WithRoleAssignments` API and assign built-in roles as shown in the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.AddAzureSearch("search");
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithRoleAssignments(search, SearchBuiltInRole.SearchIndexDataReader)
+                 .WithReference(search);
+```
+
+When you use the `WithRoleAssignments` method, it replaces the default role assignments with the specified ones. This method requires two parameters: the resource to which the role assignment applies and the built-in role to assign. In the preceding example, the `search` resource is assigned the <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataReader?displayProperty=nameWithType> role.
+
+When you replace the default role assignments with the `SearchIndexDataReader` role, the API project is restricted to only reading data from the Azure Search resource. This ensures the API project can't write data to the Azure Search resource.
+
+For more information, see [Azure built-in roles](/azure/role-based-access-control/built-in-roles).
+
+## Built-in role assignment reference
```

Comment:
Is there not somewhere in Azure we could put these? This doesn't seem like our list to maintain.